src/main/java/de/unijena/cheminf/clustering/desccalc/Descriptor.java

```diff
@@ -920,7 +1924,926 @@ private static synchronized void setWienerNumber(
+                tmpNewBond.setAtoms(new IAtom[]{tmpAtom1, tmpAtom2});
+
+                // Copy bond flags
+                if (tmpBond.isAromatic()) {
+                    tmpNewBond.setIsAromatic(true);
+                }
+                if (tmpBond.isInRing()) {
+                    tmpNewBond.setIsInRing(true);
+                }
+
+                // Copy bond properties
+                for (Object key : tmpBond.getProperties().keySet()) {
+                    tmpNewBond.setProperty(key, tmpBond.getProperty(key));
+                }
+
+                // Add bond to new container
+                tmpMoleculeWithExplicitH.addBond(tmpNewBond);
+            }
+
+            // Now that we have a copy of the original molecule, add explicit hydrogens
+            try {
+                AtomContainerManipulator.convertImplicitToExplicitHydrogens(tmpMoleculeWithExplicitH);
+            } catch (Exception anException) {
+                Descriptor.LOGGER.log(
+                        Level.SEVERE,
+                        "Descriptor.createMoleculeWithExplicitHydrogens: Error converting implicit to explicit hydrogens: " + anException.getMessage(),
+                        anException
+                );
+                throw new Exception("Failed to convert implicit to explicit hydrogens: " + anException.getMessage());
+            }
+
+            return tmpMoleculeWithExplicitH;
+
+        } catch (Exception anException) {
+            if (!(anException instanceof IllegalArgumentException)) {
+                Descriptor.LOGGER.log(
+                        Level.SEVERE,
+                        "Descriptor.createMoleculeWithExplicitHydrogens: Error creating molecule with explicit hydrogens: " + anException.getMessage(),
+                        anException
+                );
+            }
+            throw anException;
+        }
+    }
+
+    /**
+     * Uses a specified aromaticity model to modify aMolecule.
+     * Note: This method changes the input molecule by applying the specified aromaticity model.
+     * Note: No checks are performed, all necessary checks have already been made in public methods above.
+     *
+     * @param aMolecule Molecule that will be modified (IS CHANGED)
+     * @param anAromaticityModel The aromaticity model that will be used for aromaticity detection
+     * @throws Exception Thrown if the aromaticity detection fails or if the aromaticity model is null
+     */
+    public static void setAromaticity(IAtomContainer aMolecule, ElectronDonation anAromaticityModel) throws Exception {
+        try {
+            // Check if anAromaticityModel is null
+            if (anAromaticityModel == null) {
+                Descriptor.LOGGER.log(
+                        Level.SEVERE,
+                        "Descriptor.setAromaticity: Aromaticity model cannot be null."
+                );
+                throw new IllegalArgumentException("Aromaticity model cannot be null.");
```

Comment:
adding to the comment above: in general, you either log an exception and then go on or you throw an exception and let the calling code handle/log it.